object_storage_api/repositories/image.py

```diff
@@ -130,23 +131,27 @@ def update(self, image_id: str, image: ImageIn, update_primary: bool, session: C
             exc.response_detail = "Image not found"
             raise exc
 
-        stored_image = self.get(str(image_id), session=session)
-        if image.file_name != stored_image.file_name and self._is_duplicate(
-            image.entity_id, image.code, image_id, session=session
-        ):
-            raise DuplicateRecordError("Duplicate image found within the parent entity", entity_type="image")
-
-        logger.info("Updating image metadata with ID: %s", image_id)
-        if update_primary:
-            bulkwrite_update = [
-                UpdateMany(filter={"primary": True, "entity_id": image.entity_id}, update={"$set": {"primary": False}}),
-                UpdateOne(filter={"_id": image_id}, update={"$set": image.model_dump(by_alias=True)}),
-            ]
-            self._images_collection.bulk_write(bulkwrite_update, session=session)
-        else:
-            self._images_collection.update_one(
-                {"_id": image_id}, {"$set": image.model_dump(by_alias=True)}, session=session
-            )
+        try:
+            if update_primary:
+                bulkwrite_update = [
+                    UpdateOne(
+                        filter={"_id": image_id}, update={"$set": image.model_dump(by_alias=True, exclude="primary")}
```

Comment:
Will updates to other fields be made if the new name is found to be a duplicate?

Reply:
I believe this is what is means for the operations being atomic (first paragraph of https://www.mongodb.com/docs/manual/core/write-operations-atomicity/#atomicity-and-transactions)

Reply:
Yes for the `UpdateOne`. In that case is there any point in excluding the `primary` field? If it fails due to a duplicate then `primary` will not get updated, and the `UpdateMany` will not run too.

Reply:
Technically we can leave it in, but then it will set it to True, then False then True again (For the UpdateOne, UpdateMany, UpdateOne), this current way avoids touching it so if it started false and is being changed to true, it only gets modified for that entity once at the end. We cant combine the first and last UpdateOne as then the UpdateMany would reset it to false.

Reply:
Yeah what you have done is exactly what I was trying to suggest. Cheers!